src/Common/src/CoreLib/System/Globalization/DateTimeFormat.cs

```diff
@@ -458,8 +458,8 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
             }
 
             // This is a flag to indicate if we are format the dates using Hebrew calendar.
-            bool isHebrewCalendar = (cal.ID == CalendarId.HEBREW);
-            bool isJapaneseCalendar = (cal.ID == CalendarId.JAPAN);
+            bool isHebrewCalendar = !GlobalizationMode.Invariant && ((CalendarId)cal.ID == CalendarId.HEBREW);
+            bool isJapaneseCalendar = !GlobalizationMode.Invariant && ((CalendarId)cal.ID == CalendarId.JAPAN);
             // This is a flag to indicate if we are formating hour/minute/second only.
```

Comment:
When changing the definition of `GlobalizationMode.Invariant` into a constant

```
internal const bool Invariant = false;
```

this allows the C# compiler to eliminate the dead code.